cpp/src/parquet/encoding.cc

```diff
@@ -1580,10 +1660,19 @@ class DictDecoderImpl : public DecoderImpl, virtual public DictDecoder<Type> {
     if (num_values != idx_decoder_.GetBatch(indices_buffer, num_values)) {
       ParquetException::EofException();
     }
-    auto binary_builder = checked_cast<::arrow::BinaryDictionary32Builder*>(builder);
-    PARQUET_THROW_NOT_OK(binary_builder->AppendIndices(indices_buffer, num_values));
-    num_values_ -= num_values;
-    return num_values;
+
+    // It looks like this method is only called by ByteArray types. Previously,
```

Comment:
ditto, I think indices is not 64bit offset values.

Reply:
Is any change necessary here?

Reply:
No need, just ignore it